climate_categories/_conversions.py

```diff
@@ -622,50 +633,40 @@ def __init__(
         )
 
     @classmethod
-    def _read_csv_meta(cls, reader: csv.reader) -> typing.Dict[str, str]:
+    def _read_csv_meta(cls, fd: typing.TextIO):
         """Read the metadata section of a CSV conversion specification file. It consists
-        of key, value pairs, one pair on each line. A single empty line terminates the
-        metadata section.
+        of YAML key, value pairs, one pair on each line separated by a colon.
+        Each line is prefixed with the comment char "#".
 
         Parameters
         ----------
-        reader: a CSV reader object as returned by csv.reader
-            Use a CSV reader object which was not used before to read from. The reader
-            object will be iterated up to the end of the meta data section, so that
-            after calling _read_csv_meta you can directly start reading the data
-            section.
+        fd: a CSV file object
+            Use a file object which was not used before to read from.
+            The file object will be iterated up to the end of the meta data
+            section, so that after calling _read_csv_meta you can directly
+            start reading the data section.
 
         Returns
         -------
         meta_data: dict
-            lineno Mapping of meta data keys to values.
+            Mapping of meta data keys to values.
         """
         meta_data = {}
-        for row in reader:
-            if not row:
-                break
-            if len(row) < 2:
-                raise ValueError(
-                    f"Meta data specification is incomplete in line {reader.line_num}:"
-                    f" {row!r}."
-                )
-            if len(row) > 2:
-                raise ValueError(
-                    f"Meta data specification has extraneous fields in line"
-                    f" {reader.line_num}:"
-                    f" {row!r}, did you forget to escape a comma?"
-                )
-            if row[0] not in cls._meta_data_keys:
-                raise ValueError(
-                    f"Unknown meta data key in line {reader.line_num}: {row[0]}."
-                )
+        yaml_header = ""
+        last_pos = fd.tell()
+        line = fd.readline()
+        while line.startswith("#"):
+            # remove leading comment and whitespace
+            yaml_header += line[1:].lstrip()
+            last_pos = fd.tell()
+            line = fd.readline()
+        fd.seek(last_pos)
+        print(yaml_header)
+        meta_data = sy.load(yaml_header, schema=cls._strictyaml_metadata_schema).data
+        for idx, key in enumerate(meta_data.keys()):
+            if key not in cls._meta_data_keys:
+                raise ValueError(f"Unknown meta data key in line {idx + 1}: {key}.")
```

Comment:
```suggestion
```

This is checked by strictyaml already, no?

Reply:
Yeah, it is, good catch :-)

Reply:
I've removed it.